core/src/main/java/com/graphhopper/storage/index/QueryResult.java

```diff
@@ -49,8 +51,10 @@ public QueryResult(double queryLat, double queryLon) {
     }
 
     /**
-     * @return the closest matching node. -1 if nothing found, this should be avoided via a call of
-     * 'isValid'
+     * Returns the closest matching node. This is either a tower node of the base graph
+     * or a virtual node (see also {@link com.graphhopper.routing.QueryGraph#lookup(List)}).
```

Comment:
But shouldn't the javadoc precisely say what this method returns in which situations? Otherwise developers like me don't know what to expect.